docs/modules/add_extra_fields_to_client_events_unsigned.md

```diff
@@ -0,0 +1,26 @@
+# Add extra fields to client events unsigned section callbacks
+
+_First introduced in Synapse v1.96.0_
+
+This callback allows modules to add extra fields to the unsigned section of
+events when they get sent down to clients.
+
+These get called *every* time an event is to be sent to clients, so care should
+be taken to ensure with respect to performance.
+
+### API
+
+To register the callback, use
+`register_add_extra_fields_to_unsigned_client_event_callbacks` on the
+`ModuleApi`.
+
+The callback should be of the form
+
+```python
+async def add_field_to_unsigned(
+    event: EventBase,
+) -> JsonDict:
+```
```

Comment:
Thoughts:

- the module doesn't know who the event is destined for. Would it be useful to include a receipient mxid/device here? (I'm thinking of e.g. txn ids, which we only send down to the device that requested it). I expect the answer is "we don't need it now, but we can add it in the future if it's necessary".
- Could the callback return different things in different invocations for the same event? If so---do we care/need to worry?

Reply:
1. I think it could be useful in the future, but it's not needed for now and I don't really want to have to do the refactoring to support it right now. As you say, its easy to add later.
2. Yes, the callback can give different things for the same event, but that doesn't matter